client/packages/invoices/src/OutboundShipment/DetailView/SidePanel/PricingSection.tsx

```diff
@@ -141,6 +149,52 @@ const ItemPrices = ({ pricing, isDisabled }: PricingGroupProps) => {
   );
 };
 
+export const ForeignCurrencyPrices = ({
+  pricing,
+  currency,
+  otherPartyIsInternal,
+  currencyRate,
+  onChange,
+}: CurrencyPricingProps) => {
+  const t = useTranslation('distribution');
+  const { store } = useAuthContext();
+  const { c: foreignCurrency } = useCurrency(currency?.code as Currencies);
+
+  return (
+    <>
+      <PanelRow style={{ marginTop: 12 }}>
+        <PanelLabel fontWeight="bold">
+          {t('heading.foreign-currency')}
+        </PanelLabel>
+        <PanelField>
+          <CurrencyModal
+            onChange={onChange}
+            isDisabled={
+              otherPartyIsInternal || !store?.preferences.issueInForeignCurrency
+            }
+            currency={currency as CurrencyRowFragment}
+            currencyRate={currencyRate ?? 1}
```

Comment:
Might be late in the game but... can we make currency guaranteed? If the answer is yes I'd do it in another issue. All invoices SHOULD at least have the default currency as default currency, so shouldn't be optional really.

Reply:
#3134